cpp/src/svm/svm_api.cpp

```diff
@@ -152,13 +192,46 @@ cumlError_t cumlDpSvcFit(cumlHandle_t handle,
                       kernel_param,
                       model,
                       static_cast<double*>(nullptr));
-      *n_support     = model.n_support;
-      *b             = model.b;
-      *dual_coefs    = model.dual_coefs;
-      *x_support     = model.support_matrix.data;
-      *support_idx   = model.support_idx;
-      *n_classes     = model.n_classes;
-      *unique_labels = model.unique_labels;
+      *n_support = model.n_support;
+      *b         = model.b;
+      *n_classes = model.n_classes;
+      if (model.dual_coefs->size() > 0) {
+        *dual_coefs = (double*)rmm_alloc.allocate_async(
+          model.dual_coefs->size(), rmm::CUDA_ALLOCATION_ALIGNMENT, stream);
```

Comment:
As mentioned in https://github.com/rapidsai/cuml/issues/6057 by @harrism , these raw calls to `allocate` need to go away. Should we change this API to assume that the user is passing in pre-allocated memory?

Reply:
Hmmm that is a tricky situation. @harrism any thoughts?

Reply:
Use a device_buffer?

Reply:
Oh, this is a C API? My solution would be to not have a C API!

Reply:
C API source code is written in `.cpp` and we just export the API in `.h` header via `extern C`